Sources/SwiftOperatorPrecedence/OperatorPrecedenceError+Diagnostics.swift

```diff
@@ -0,0 +1,103 @@
+//===------------------ OperatorPrecedenceError.swift ---------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftParser
+import SwiftSyntax
+
+extension OperatorPrecedenceError : DiagnosticMessage {
+  public var severity: DiagnosticSeverity {
+    .error
+  }
+
+  public var message: String {
+    switch self {
+    case .groupAlreadyExists(let existing, _):
+      return "redefinition of precedence group '\(existing.name)'"
+
+    case .missingGroup(let groupName, _):
+      return "unknown precedence group '\(groupName)'"
+
+    case .operatorAlreadyExists(let existing, _):
+      return "redefinition of infix operator '\(existing.name)'"
+
+    case .missingOperator(let operatorName, _):
+      return "unknown infix operator '\(operatorName)'"
+
+    case .incomparableOperators(_, let leftGroup, _, let rightGroup):
+      if leftGroup == rightGroup {
+        return "adjacent operators are in non-associative precedence group '\(leftGroup)'"
+      }
+
+      return "adjacent operators are in unordered precedence groups '\(leftGroup)' and '\(rightGroup)'"
+    }
+  }
+
+  /// A string representation of each case.
+  private var diagnosticCaseID: String {
```

Comment:
Is there any particular reason why you want to manually specify these names instead of deriving them using reflection like we do in the parser?

Reply:
No reason; I'll switch to reflection.